src/client_builder.rs

```diff
@@ -61,49 +160,85 @@ where
+    pub fn tls_kind(&mut self, kind: TlsKind) -> &mut Self {
+        self.tls_kind = kind;
         self
     }
 
-    /// Return a new [`Client`] using a `native-tls` transport.
-    #[cfg(feature = "native-tls")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "native-tls")))]
-    pub fn native_tls(&mut self) -> Result<Client<TlsStream<TcpStream>>> {
-        self.connect(|domain, tcp| {
-            let ssl_conn = TlsConnector::builder().build()?;
-            Ok(TlsConnector::connect(&ssl_conn, domain, tcp)?)
-        })
+    /// Controls the use of certificate validation.
+    ///
+    /// Defaults to `false`.
+    ///
+    /// # Warning
+    ///
+    /// You should only use this as a last resort as it allows another server to impersonate the
+    /// server you think you're talking to, which would include being able to receive your
+    /// credentials.
+    ///
+    /// See [`native_tls::TlsConnectorBuilder::danger_accept_invalid_certs`],
+    /// [`native_tls::TlsConnectorBuilder::danger_accept_invalid_hostnames`],
+    /// [`rustls::ClientConfig::dangerous`]
+    #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+    pub fn danger_skip_tls_verify(&mut self, skip_tls_verify: bool) -> &mut Self {
+        self.skip_tls_verify = skip_tls_verify;
+        self
     }
 
-    /// Return a new [`Client`] using `rustls` transport.
-    #[cfg(feature = "rustls-tls")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "rustls-tls")))]
-    pub fn rustls(&mut self) -> Result<Client<RustlsStream<TcpStream>>> {
-        self.connect(|domain, tcp| {
-            let ssl_conn = RustlsConnector::new_with_native_certs()?;
-            Ok(ssl_conn.connect(domain, tcp)?)
-        })
+    /// Make a [`Client`] using the configuration.
+    ///
+    /// ```no_run
+    /// # use imap::ClientBuilder;
+    /// # {} #[cfg(feature = "rustls-tls")]
+    /// # fn main() -> Result<(), imap::Error> {
+    /// let client = ClientBuilder::new("imap.example.com", 143)
+    ///     .starttls().connect()?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn connect(&self) -> Result<Client<Connection>> {
+        #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+        return self.connect_with(|_domain, tcp| self.build_tls_connection(tcp));
+        #[cfg(all(not(feature = "native-tls"), not(feature = "rustls-tls")))]
+        return self.connect_with(|_domain, _tcp| -> Result<Connection> {
+            return Err(Error::TlsNotConfigured);
+        });
     }
 
-    /// Make a [`Client`] using a custom TLS initialization. This function is intended
+    /// Make a [`Client`] using a custom initialization. This function is intended
     /// to be used if your TLS setup requires custom work such as adding private CAs
     /// or other specific TLS parameters.
     ///
+    /// Note, if the connection does not end up as TLS, handshake will not be called.
+    /// This can happen if connection mode is set to Tcp, or the server does not support starttls.
```

Comment:
```suggestion
    /// This can happen if connection mode is set to "plaintext", or if the server does not support STARTTLS (and the connection mode is `Auto`).
```